modules/core/src/main/scala/gem/Dataset.scala

```diff
@@ -35,7 +36,7 @@ object Dataset {
         case -1 => None
         case  n =>
           val (a, b) = s.splitAt(n)
-          b.drop(1).parseInt.toOption.flatMap { n =>
+          MonadError[Either[Throwable, ?], Throwable].catchNonFatal(b.drop(1).toInt).toOption.flatMap { n =>
             Observation.Id.fromString(a).map(oid => Dataset.Label(oid, n))
```

Comment:
Yeah I want to PR those into mouse, which is where these things are ending up. I did this for expediency but we need to fix it.